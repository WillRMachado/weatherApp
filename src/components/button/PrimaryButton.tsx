import React from 'react';
import {
  StyleSheet,
  TouchableOpacity,
  TouchableOpacityProps,
} from 'react-native';
import {measures} from '../../styles/index';
import {useTheme} from '@react-navigation/native';
import {themeColorsTypes} from '../../styles/themes/themesType';
import TranslateText from '../translate/TranslateText';

const PrimaryButton = (props: {text: string} & TouchableOpacityProps) => {
  const {text} = props;
  const {colors}: themeColorsTypes = useTheme();
  const styles = dynamicStyles(colors);

  return (
    <TouchableOpacity style={styles.buttonContainer} {...props}>
      <TranslateText string={text} style={styles.buttonText} />
    </TouchableOpacity>
  );
};

export default PrimaryButton;

const dynamicStyles = (colors: themeColorsTypes) =>
  StyleSheet.create({
    buttonContainer: {
      backgroundColor: colors.secondary,
      alignItems: 'center',
      borderWidth: measures.border.borderWidthStandard,
      elevation: 8,
      borderRadius: measures.paddingAdjustedScreenWidth,
      margin: measures.standardPadding,
    },
    buttonText: {
      fontSize: measures.fontSize.M,
      padding: measures.standardPadding / 2,
      paddingLeft: measures.standardPadding,
      paddingRight: measures.standardPadding,
      color: colors.primary,
    },
  });
