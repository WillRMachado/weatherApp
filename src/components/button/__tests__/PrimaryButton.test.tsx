import React from 'react';
import {TouchableOpacity} from 'react-native';
import {act} from 'react-test-renderer';
import screenRender from '../../../../jestHelpers/screenRender';
import screenSnapshotCreator from '../../../../jestHelpers/screenSnapshotCreator';
import PrimaryButton from '../PrimaryButton';

describe('PrimaryButton Component', () => {
  const onPressCallback = jest.fn(() => {});
  const PrimaryButtonWithProps = () => (
    <PrimaryButton text={'weather.unableToFind'} onPress={onPressCallback} />
  );

  it('renders correctly', async () => {
    const component = await screenRender(PrimaryButtonWithProps);
    expect(component).toBeTruthy();
  });

  it('matches snapshot', async () => {
    const component = await screenSnapshotCreator(PrimaryButtonWithProps);
    expect(component).toMatchSnapshot();
  });

  it('clicks', async () => {
    const component = await screenRender(PrimaryButtonWithProps);
    const button = component.root.findAllByType(TouchableOpacity)[0];
    act(() => {
      button.props.onPress();
    });
    expect(onPressCallback.mock.calls.length).toBe(1);
  });
});
