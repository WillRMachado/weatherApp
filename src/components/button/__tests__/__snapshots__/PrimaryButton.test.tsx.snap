// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PrimaryButton Component matches snapshot 1`] = `null`;
